src/analysis/scoping_analysis.cpp

```diff
@@ -130,8 +133,18 @@ struct ScopingAnalysis::ScopeNameUsage {
     StrSet got_from_closure;
     StrSet passthrough_accesses; // what names a child scope accesses a name from a parent scope
 
+    // `import *` and `exec` both force the scope to use the NAME lookup
+    bool hasNameForcingSyntax;
+    AST_ImportFrom* nameForcingNodeImportStar;
+    AST_Exec* nameForcingNodeBareExec;
+
+    // If it has a free variable / if any child has a free variable
+    bool free;
+    bool child_free;
```

Comment:
Are these equivalent to `(bool)got_from_closure.size()` and `(bool)passthrough_accesses.size()`?